packages/markdown/remark/src/rehype-collect-headers.ts

```diff
@@ -1,12 +1,14 @@
 import { visit } from 'unist-util-visit';
-import slugger from 'github-slugger';
+import Slugger from 'github-slugger';
+
+import type { MarkdownHeader, RehypePlugin } from './types.js';
 
-/**  */
 export default function createCollectHeaders() {
-	const headers: any[] = [];
+	const headers: MarkdownHeader[] = [];
+	const slugger = new Slugger();
 
-	function rehypeCollectHeaders() {
-		return function (tree: any) {
+	const rehypeCollectHeaders: RehypePlugin<[]> = () => {
```

Comment:
style nit: we usually avoid `const X = () => {}` in this code base, in favor of `function X() {}`. 

Unless I'm missing something, you should be able to get the same type info like this:

```
function rehypeCollectHeaders(): RehypePlugin<[]> {
```

Reply:
Updated that!